awd/cli.py

```diff
@@ -183,32 +190,19 @@ def deposit(
                     e.response["Error"]["Message"],
                 )
 
-            # Retrieve and validate Crossref metadata
-            crossref_response = crossref.get_response_from_doi(metadata_url, doi)
-            if crossref.is_valid_response(doi, crossref_response) is False:
-                continue
-            article.crossref_metadata = crossref.get_metadata_extract_from(
-                crossref_response.json()
-            )
-
-            # Create and validate DSpace metadata from Crossref metadata
-            article.dspace_metadata = crossref.create_dspace_metadata_from_dict(
-                article.crossref_metadata,
-                "config/metadata_mapping.json",
-            )
-            if crossref.is_valid_dspace_metadata(article.dspace_metadata) is False:
+            try:
+                article.process()
+            except (
+                InvalidArticleContentResponseError,
+                InvalidCrossrefMetadataError,
+                InvalidDSpaceMetadataError,
+            ):
```

Comment:
Loving it!  I can assume if these exceptions are thrown, it's likely a handled / expected exception, and this DOI will just be skipped (and FWIW, here too could be additional logic to handle those).  But if anything _unhandled_ slips thorough, it will bubble up to the CLI function level.